roles/gitlab/tasks/configure.yml

```
@@ -19,6 +19,32 @@
     - "Reconfigure Primary GitLab"
     - "Reconfigure Non Primary GitLab"
 
+- name: "Ensure gitaly['configuration'] is not present in gitlab_additional_configurations"
+  when: "gitlab_use_internal_gitaly"
+  block:
+    - name: "Check if gitaly is present"
+      when:
+        - "gitlab_additional_configurations | length > 0"
+        - "item.gitaly is defined"
+      ansible.builtin.set_fact:
+        __gitaly_exists: true
+        __gitaly_item: "{{ item.gitaly }}"
+      with_items: "{{ gitlab_additional_configurations }}"
+
+    - name: "Check if gitaly['configuration'] is present"
+      when:
+        - "item.key is defined"
+        - "item.key == 'configuration'"
+      ansible.builtin.set_fact:
+        __gitaly_configuration_exists: true
+      with_items: "{{ __gitaly_item | default([]) }}"
```

Could this also work?

```suggestion
    - name: "Check if gitaly['configuration'] is present"
      ansible.builtin.set_fact:
        __gitaly_configuration_exists: "{{ gitlab_additional_configurations | selectattr('gitaly.configuration', 'defined')| list | length > 0 }}"
```

---

I don't think this will work since the structure of the variable looks like this:

```yaml
gitlab_additional_configurations:
  - gitaly:
    - key: 'configuration'
      type: 'plain'
      value: '...'
```

Due to the nested lists and the `configuration` parameter hidden behind `key` I came up with this relatively complex implementation.

---

I'm sure there must be a shorter way, but let's leave it that way for now.